### PR TITLE
[Resolve #444]FAQ of one synchronization issue

### LINKED ISSUE #444: Qitmeer loses peers and refuse to connect new peers
### version
0.9.2+dev-d56e01e, latest on filing this issue

#### steps
at the short period after qitmeer launching, nodes running well in synchronization.

```sh
2021-03-16|16:38:13.081 [INFO ] RPC server listening on             addr=0.0.0.0:18131
2021-03-16|16:38:13.132 [INFO ] Loaded 16 addresses from file '/home/tony/.qitmeerd/data/testnet/peers.json'
2021-03-16|16:38:13.132 [INFO ] Connection manager started
2021-03-16|16:38:13.153 [INFO ] P2P Server listening on             addr=0.0.0.0:18130
2021-03-16|16:38:13.172 [INFO ] 1 addresses found from DNS seed testnet-seed.hlcwallet.info
2021-03-16|16:38:13.398 [INFO ] New valid peer: 103.29.70.78:18130 (outbound),user-agent:qitmeer:0.9.2 module=blkmanager
2021-03-16|16:38:13.399 [INFO ] Syncing to state (823997,761664,761664,823998,1) from peer 103.29.70.78:18130 cur graph state:(228799,221724,221724,228800,1) module=blkmanager
2021-03-16|16:38:13.608 [INFO ] New valid peer: 113.52.132.171:18130 (outbound),user-agent:qitmeer:0.9.2 module=blkmanager
2021-03-16|16:38:13.809 [INFO ] 1 addresses found from DNS seed seed.qitmir.info
2021-03-16|16:38:14.165 [INFO ] 4 addresses found from DNS seed testnet-seed.qitmeer.xyz
2021-03-16|16:38:21.675 [INFO ] Processed 1 block in the last 1m23.36s (1 transaction, order 228800, 2020-09-09 17:37:26 +0800 CST) module=blkmanager
2021-03-16|16:38:25.102 [INFO ] Syncing to state (823997,761664,761664,824135,1) from peer 113.52.132.171:18130 cur graph state:(229299,222197,222197,229300,1) module=blkmanager
```

and I can find a few peers
```sh
curl --data '{"method":"getPeerInfo","params":[],"jsonrpc":"2.0","id":1}' -s -k -u "test:test"  -H 'Content-Type: application/json' http://127.0.0.1:18131 | jq |grep addr\"

"addr": "103.29.70.78:18130",
      "addr": "47.103.61.128:18130",
      "addr": "103.231.255.228:18130",

```

however, peers keep disconnecting

```
2021-03-16|16:59:56.357 [INFO ] Peer 47.103.61.128:18130 (outbound) appears to be stalled or misbehaving, inv timeout -- disconnecting
2021-03-16|16:59:56.804 [INFO ] Lost peer                           module=blkmanager peer="47.103.61.128:18130 (outbound)"
```

and the new peers disconnect immediately after being connected.

```sh
2021-03-16|17:03:45.778 [INFO ] New valid peer: 103.29.70.78:18130 (outbound),user-agent:qitmeer:0.9.2 module=blkmanager
2021-03-16|17:04:38.230 [INFO ] New valid peer: 103.231.255.228:18130 (outbound),user-agent:qitmeer:0.9.2 module=blkmanager
2021-03-16|17:04:42.014 [INFO ] New valid peer: 47.103.61.128:18130 (outbound),user-agent:qitmeer:0.9.2 module=blkmanager
2021-03-16|17:05:55.106 [INFO ] Syncing to state (824052,761712,761712,824053,1) from peer 103.231.255.228:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
2021-03-16|17:05:58.092 [INFO ] Syncing to state (824052,761712,761712,824053,1) from peer 103.231.255.228:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
2021-03-16|17:05:58.178 [INFO ] Lost peer                           module=blkmanager peer="103.231.255.228:18130 (outbound)"
2021-03-16|17:05:58.178 [INFO ] Syncing to state (824052,761712,761712,824054,1) from peer 47.103.61.128:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
2021-03-16|17:06:01.084 [INFO ] Syncing to state (824052,761712,761712,824054,1) from peer 47.103.61.128:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
2021-03-16|17:06:01.217 [INFO ] Lost peer                           module=blkmanager peer="47.103.61.128:18130 (outbound)"
2021-03-16|17:06:01.217 [INFO ] Syncing to state (824052,761712,761712,824053,1) from peer 103.29.70.78:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
2021-03-16|17:06:04.120 [INFO ] Syncing to state (824052,761712,761712,824053,1) from peer 103.29.70.78:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
2021-03-16|17:06:04.228 [INFO ] Lost peer                           module=blkmanager peer="103.29.70.78:18130 (outbound)"

```

then the synchronization got stuck at block 235979
```sh
2021-03-16|17:06:04.266 [INFO ] Syncing to state (824024,761687,761687,824025,1) from peer 47.116.50.38:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
2021-03-16|17:06:07.080 [INFO ] Syncing to state (824024,761687,761687,824025,1) from peer 47.116.50.38:18130 cur graph state:(235978,228535,228535,235979,1) module=blkmanager
```

and there are no peers connected
```sh
 $ curl --data '{"method":"getPeerInfo","params":[],"jsonrpc":"2.0","id":1}' -s -k -u "test:test"  -H 'Content-Type: application/json' http://127.0.0.1:18131 | jq                                                                           

{
  "jsonrpc": "2.0",
  "id": 1,
  "result": []
}

```

### P.S.
1. Having watching the CPU and memory,  no apparent sign of exceptions
2. Reproduced at least 5 times.

### PR DESCRIPTION
**Pull request by forchain**
https://github.com/Qitmeer/qitmeer/issues/444

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -190,6 +190,17 @@ If you somehow need to stick to the current version, please make sure to execute
 ```shell
 go mod tidy
 ```
+### Qitmeer gets stuck on synchronization.
```
Comment (dindinw) [inline README.md:193]: Please add more note to address that the issue can't generally reproduce and is 0.9.x only.

File: README.md [hunk 2]
```diff
@@ -190,6 +190,17 @@ If you somehow need to stick to the current version, please make sure to execute
...
+For some reasons, all your peers are disconnected. For example, all your peers are busy, so your synchronization
+requests to them would exceed the timeout, leading your nodes disconnection from those peers. One recommended
+practice to mitigate this issue is to enhance your connectivity by reporting your external ip (if you have) to the public, 
+so new peers will reach to you even all your peers are disconnected.
+Please see Qitmeer/qitmeer/issues/444 to get more details
```
Comment (dindinw) [inline README.md:198]: Please fix the link

**Review by dindinw — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (forchain) [reply README.md]: fixed https://github.com/Qitmeer/qitmeer/pull/445/commits/52744e4f9de6fb2d370d9ab0c0491a6110434c08

File: README.md (hunk 1 above)
Comment (forchain) [reply README.md:201]: updated, adjusted according the latest recommended memory setting
https://github.com/Qitmeer/qitmeer/pull/445/commits/52744e4f9de6fb2d370d9ab0c0491a6110434c08

**Review by dindinw — APPROVED** [review]

**Pull request merged**